[AVAX] Merge Cortina 8 (v1.10.8)

evlekht opened the pull request:
## Why this should be merged
This PR contains changes from ava-labs/avalanchego between cortina-7 and cortina-8.
https://github.com/ava-labs/avalanchego/releases/tag/v1.10.8
### Conflicts
- .github/workflows/debian/template/control
- RELEASES.md
- genesis/genesis_local.json
- genesis/genesis_test.go
- go.sum
- scripts/build_test.sh
- scripts/mocks.mockgen.txt
- version/compatibility.json
- version/constants.go
- version/string.go
- vms/platformvm/state/mock_state.go
- vms/platformvm/state/state.go
### Post-merge fixes
#### Updated genesis IDs
Caminogo doesn't have avax mainnet and fuji genesis configs, so when `GetConfig` is called with one of those ids, it returns local genesis with a modified networkID.

`genesis/genesis_local.json` start time was updated in Cortina 8. 

Because of these two things, all three genesis hashes had changed: avax mainnet, fuji and local.
#### PVM block-builder/network
Starting from cortina 8, `network.blockBuilder` is `*builder`, not a `Builder` interface. Cause of that, we no longer can put caminoBuilder there, so we're directly referencing avax block builder that is wrapped in camino builder.
#### gomock.Controller.Finish() removal
Starting from go 1.14, if controller is created with `*testing.T`, its no longer needed to call `.Finish()` on controller.
See https://github.com/ava-labs/avalanchego/pull/1803
This PR removes similar code for consistency.
## How this was tested
With already existing unit-tests and e2e tests.

Commit:
Remove calls to `ctrl.Finish` (#1803)

Co-authored-by: Sam Batschelet <sam.batschelet@avalabs.org>

maru-ava committed:
e2e: Remove unnecessary transaction status checking (#1786)

dhrubabasu committed:
Fix p2p mockgen location (#1806)

Commit:
fix end proof verification (#1801)

Commit:
`merkledb` -- add proof fuzz test (#1804)

Co-authored-by: Sam Batschelet <sam.batschelet@avalabs.org>

Commit:
`sync` -- re-add network client metrics (#1787)

joshua-kim committed:
Add function to initialize set from elements (#1808)

dboehm-avalabs committed:
Add Maybe to the end bound of proofs (Part 1) (#1793)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>

amirhasanzadehpy committed:
Add go version to `--version` output (#1819)

maru-ava committed:
e2e: Add local network fixture (#1700)

StephenButtolph committed:
Fix test flake in TestProposalTxsInMempool (#1822)

Commit:
`sync` -- remove todo (#1788)

dboehm-avalabs committed:
Add Maybe to the end bound of proofs (Part 2) (#1813)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>

Commit:
Move Maybe to its own package (#1817)

Commit:
`merkledb` -- clarify/improve change proof invariants (#1810)

dhrubabasu committed:
P-chain state prune + height index (#1719)

StephenButtolph committed:
Update maintainer of the debian packages (#1825)

dhrubabasu committed:
Make platformvm implement `block.HeightIndexedChainVM` (#1746)

dhrubabasu committed:
Add P-chain `GetBlockByHeight` API method (#1747)

ceyonur committed:
Update local genesis startTime (#1811)

Commit:
`sync` -- add handling for fatal error (#1690)

Co-authored-by: Darioush Jalali <darioush.jalali@avalabs.org>

StephenButtolph committed:
Add error logs for unexpected proposervm BuildBlock failures (#1832)

StephenButtolph committed:
Fix subnet validator set public key initialization (#1833)

StephenButtolph committed:
Document PendingTxs + BuildBlock consensus engine requirement (#1835)

dependabot[bot] committed:
Bump github.com/supranational/blst from 0.3.11-0.20230406105308-e9dfc5ee724b to 0.3.11 (#1831)

abi87 committed:
Add Primary Network Partial Sync Option (#1769)

Co-authored-by: Stephen Buttolph <stephen@avalabs.org>

StephenButtolph committed:
Check P-chain ShouldPrune during Initialize (#1836)

StephenButtolph committed:
Update version for v1.10.8 (#1837)

knikos commented on vms/platformvm/txs/executor/standard_tx_executor.go:
```diff
@@ -309,6 +320,14 @@ func (e *StandardTxExecutor) AddValidatorTx(tx *txs.AddValidatorTx) error {
+			zap.Stringer("nodeID", tx.Validator.NodeID),
+		)
+	}
 	return nil
```
with regards to this warning log and taking into consideration https://github.com/ava-labs/avalanchego/pull/1769 and the rationale behind this logs, I believe we should add a similar warning log in our `vms/platformvm/txs/executor/camino_tx_executor.go` and more specifically in the AddValidatorTx implementation.

evlekht replied:
good point, added

knikos commented on api/admin/service_test.go:
```diff
@@ -100,7 +98,6 @@ func TestLoadVMsGetAliasesFails(t *testing.T) {
 	require := require.New(t)
 
 	resources := initLoadVMsTest(t)
-	defer resources.ctrl.Finish()
```
we should refactor our test cases to reflect this change according to https://github.com/ava-labs/avalanchego/pull/1803. I guess we should create a different issue for that to keep this PR as minimal as possible

evlekht replied:
Created issue
https://github.com/chain4travel/camino-private-issues/issues/366

knikos commented on .github/workflows/test.e2e.persistent.yml:
```diff
@@ -0,0 +1,31 @@
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.19'
```
any reason we ignore this practice from avax to run golang with latest 1.19 patch 
```yaml
          go-version: '~1.19.12'
          check-latest: true
 ```
 That applies also to all other build and test workflows as well as the static-analysis.yml

evlekht replied:
Just consistency - we didn't change form 1.19 to 1.19.6 before, so I'm keeping it this way with assumption that we had a reason for this. I'd prefer to consciously revisit this topic, when we'll do pass on workflows/scripts after cortina merges.

evlekht committed:
[AVAX] Post Merge fixes for Cortina 8 (v1.10.8)

evlekht committed:
[AVAX] Intermediate Merge Cortina 8 (v1.10.8)

# Conflicts:
#	.github/workflows/debian/template/control
#	RELEASES.md
#	genesis/genesis_local.json
#	genesis/genesis_test.go
#	go.sum
#	scripts/mocks.mockgen.txt
#	version/compatibility.json
#	version/constants.go
#	version/string.go
#	vms/platformvm/state/mock_state.go
#	vms/platformvm/state/state.go

knikos commented on tests/fixture/testnet/config.go:
```diff
@@ -273,7 +273,16 @@ func (nc *NodeConfig) EnsureNodeID() error {
+	if err != nil {
+		return fmt.Errorf("failed to parse nodeID bytes: %w", err)
+	}
+	nc.NodeID = nodeID
```
how come not just call  peer.CertToID(tlsCert.Leaf) directly?
